lib/spack/docs/build_systems/gopackage.rst

```diff
@@ -0,0 +1,164 @@
+  Go 1.14.
+
+Not surprisingly, the best approach to packaging a Go-based project
+for Spack depends on the details of the project.
+
+The ``GoPackage`` base class is intended to make it easy to package
+simple projects.  Tools & techniques that it uses may also be useful
+when using other build system base classes (e.g. ``MakefilePackage``),
+in particular:
+
+- Use the package's ``setup_build_environment`` to set
+  ``GO111MODULE=on`` to enable module mode and set
+  ``GOFLAGS="-mod-vendor"`` to enable vendoring, which prevents ``go``
+  from downloading libraries at build time.
+- If the application hasn't vendored its dependencies, use
+  ``modules2tuple`` to generate "bulk" resource statements and include
+  them in the package definition via ``import_resources``.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Simple package definitions
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The simplest case is a project that uses a simple invocation of ``go
+build`` to compile an executable, *and* uses modules, *and* has
+vendored its dependencies.  Here's an example of a package definition
+which is only slightly complicated by the fact that it requires a
+newer version of Go than ``GoPackage`` provides by default:
+
+.. code-block:: python
+
+    from spack import *
+
+
+    class Lazygit(GoPackage):
+        """Simple terminal UI for git commands."""
+
+        homepage = "https://github.com/jesseduffield/lazygit"
+        url      = "https://github.com/jesseduffield/lazygit/archive/v0.11.3.tar.gz"
+
+        version('0.11.3', sha256='b3c503de6b34fd4284fd70655e7f42feafc07f090e7f7cc00db261f56c583c46')
+
+        depends_on('go@1.13:', type='build')  # go.mod value overrides default
+
+        executables = ['lazygit']
+
+While it generally follows the common patterns of a Spack package,
+it's worth calling attention to its:
+
+* adding a *build* dependency on the version of the Go package that it
+  requires (from the existing `go.mod`);
+* using the default ``build`` and ``install`` steps that ``GoPackage``
+  provides; and
+* providing the name of the executable to be installed by the default
+  ``install`` step.
+
+The ``GoPackage`` provides a `setup_build_environment` that sets
+`GO111MODULES=on` and `GOFLAGS=-mod=vendor` and its default build step
+invokes ``go build``.  Additional arguments to ``go build`` can be
+provided by setting ``build_args``, e.g. one could
+
+.. code-block:: python
+
+        build_args = ['-p', '12']  # adjust parallelism of build cmd
+
+to invoke ``go build -p 12``.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Slightly more complicated packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Packages that can be built with a simple invocation of ``go build``,
+that use modules, *but have not vendored their dependencies* require a
+bit of extra work.  In particular, in order to satisfy Spack's
+requirement that nothing is downloaded during the build step, the
+package needs to define "resources" that describe from whence to fetch
```

Comment:
Maybe add a link to the Resources section of the Packaging Guide

Reply:
Tried this

```
:ref:`resources <resources-expanding-extra-tarballs>`
```

but it didn't do what I was hoping for (indeed, it generated an error).

Can I buy a clue?

Reply:
figured it out....